core/src/Kokkos_GraphNode.hpp

```diff
@@ -246,6 +247,63 @@ class GraphNodeRef {
                       std::forward<Functor>(functor));
   }
 
+#if defined(KOKKOS_ENABLE_CUDA) ||                                           \
+    (defined(KOKKOS_ENABLE_HIP) && defined(KOKKOS_IMPL_HIP_NATIVE_GRAPH)) || \
+    (defined(KOKKOS_ENABLE_SYCL) && defined(SYCL_EXT_ONEAPI_GRAPH))
+  /// Capture the backend operations that calling the passed function object
+  /// with the passed execution space instance enqueues into the underlying
+  /// backend queue.
+  /// @note The function object will be stored by the underlying graph
+  ///       implementation, thereby ensuring that any captured data is kept
+  ///       alive until the graph is destroyed.
+  /// @note The parameter @p exec can be used for multi-GPU graphs.
+  ///       The captured nodes will be assigned to the device associated with
+  ///       @p exec.
+  /// @warning It's the user's responsability to ensure that theere is no
+  ///          illegal operation going on while capturing.
+  /// @todo Support label.
```

Comment:
@dalg24 Is it sufficient for a draft ?

Reply:
So with this *docstring* and the example I just pushed, would it be OK ? (@dalg24 @maartenarnst)

Reply:
Let's get started with:
- https://github.com/kokkos/kokkos-core-wiki/pull/648

Reply:
And the next one is:
- https://github.com/kokkos/kokkos-core-wiki/pull/652